panel/src/components/Forms/Input/TextareaInput.vue

```diff
@@ -291,36 +299,44 @@ export default {
 				title: selection
 			};
 		},
-		prepend(text) {
-			this.insert(text + " " + this.selection());
+		async prepend(text) {
+			return this.insert(text + " " + this.selection());
 		},
+		/**
+		 * @deprecated 5.0.0 Use `setSelectionRange` instead
+		 */
 		restoreSelectionCallback() {
 			// store selection
-			const start = this.$refs.input.selectionStart;
-			const end = this.$refs.input.selectionEnd;
+			const selectionRange = this.selectionRange();
 
 			// restore selection as `insert` method
 			// depends on it
-			return (callback) => {
-				setTimeout(() => {
-					this.$refs.input.setSelectionRange(start, end);
+			return async (callback) => {
+				await this.setSelectionRange(selectionRange);
 
-					if (callback) {
-						callback();
-					}
-				});
+				if (callback) {
+					callback();
+				}
```

Comment:
`Question:` Isn't this losing now the implicit next tick that `setTimeout` would add before?

Reply:
The nextTick is now handled in restoreSelection